docs/en/Community-Articles/2025-10-10-Using-Transfer-State-with-Angular-SSR/post.md

```diff
@@ -0,0 +1,274 @@
+
+```ts
+// books.service.ts
+import {
+    Injectable,
+    PLATFORM_ID,
+    makeStateKey,
+    TransferState,
+    inject,
+} from '@angular/core';
+import { isPlatformServer } from '@angular/common';
+import { HttpClient } from '@angular/common/http';
+import { Observable, of } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+export interface Book {
+    id: number;
+    name: string;
+    price: number;
+}
+
+@Injectable({ providedIn: 'root' })
+export class BooksService {
+    BOOKS_KEY = makeStateKey<Book[]>('books:list');
+    readonly httpClient = inject(HttpClient);
+    readonly transferState = inject(TransferState);
+    readonly platformId = inject(PLATFORM_ID);
+
+    getBooks(): Observable<Book[]> {
+        // If browser and we have the data that already fetched on the server, use it and remove from TransferState
+        if (this.transferState.hasKey(this.BOOKS_KEY)) {
+            const cached = this.transferState.get<Book[]>(this.BOOKS_KEY, []);
+            this.transferState.remove(this.BOOKS_KEY); // remove to avoid stale reads
+            return of(cached);
+        }
+
+        // Otherwise fetch data. If running on the server, write into TransferState
+        return this.httpClient.get<Book[]>('/api/books').pipe(
+            tap(list => {
+                if (isPlatformServer(this.platformId)) {
+                    this.transferState.set(this.BOOKS_KEY, list);
+                }
+            })
+        );
+    }
+}
+
+```
+
+Use it in a component:
+
+```ts
+// books.component.ts
+import { Component, inject, OnInit } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { BooksService, Book } from './books.service';
+
+@Component({
+    selector: 'app-books',
+    imports: [CommonModule],
+    template: `
+    <h1>Books</h1>
+    <ul>
+      @for (book of books; track book.id) {
+        <li>{{ book.name }} — {{ book.price | currency }}</li>
+      }
+    </ul>
+  `,
+})
+export class BooksComponent implements OnInit {
+    private booksService = inject(BooksService);
+    books: Book[] = [];
+
+    ngOnInit() {
+        this.booksService.getBooks().subscribe(data => (this.books = data));
+    }
+}
+
+```
+
+Route resolver variant (keeps templates simple and aligns with SSR prefetching):
+
+```ts
+// src/app/routes.ts
+
+export const routes: Routes = [
+  {
+    path: 'books',
+    component: BooksComponent,
+    resolve: {
+      books: () => inject(BooksService).getBooks(),
+    },
+  },
+];
+```
+
+Then read `books` from the `ActivatedRoute` data in your component.
+
+---
+
+## Option B — Using HttpInterceptor to Automate TransferState
+
+Like Option A, but less boilerplate. This approach uses an **HttpInterceptor** to automatically cache HTTP GET (also POST/PUT request but not recommended) responses in TransferState. You can determine which requests to cache based on URL patterns.
+
+Example interceptor that caches GET requests:
+
+```ts
```

Comment:
Hello @erdemcaygor thank you for explaining this feature in an article. I guess the blank ts tag is added here by mistake. We can merge this after we fix tihis.